service/evaluation/evaluation.go

```diff
@@ -650,16 +650,29 @@ func (svc *Service) evaluateControl(ctx context.Context, toe *orchestrator.Targe
 	var nonCompliantAssessmentResults = []string{}
 
 	for _, r := range results {
-		if (r.Status == evaluation.EvaluationStatus_EVALUATION_STATUS_COMPLIANT ||
-			r.Status == evaluation.EvaluationStatus_EVALUATION_STATUS_COMPLIANT_MANUALLY) &&
-			(status != evaluation.EvaluationStatus_EVALUATION_STATUS_NOT_COMPLIANT &&
-				status != evaluation.EvaluationStatus_EVALUATION_STATUS_NOT_COMPLIANT_MANUALLY) {
-			status = evaluation.EvaluationStatus_EVALUATION_STATUS_COMPLIANT
-		} else if r.Status == evaluation.EvaluationStatus_EVALUATION_STATUS_PENDING &&
-			status == evaluation.EvaluationStatus_EVALUATION_STATUS_PENDING {
-			status = evaluation.EvaluationStatus_EVALUATION_STATUS_PENDING
-		} else {
-			status = evaluation.EvaluationStatus_EVALUATION_STATUS_NOT_COMPLIANT
+		// status is the current evaluation status, r.Status is the current assessment result status
+		// Note: Status should only contain the evaluation status without _MANUALLY!
+		switch status {
```

Comment:
What is the point of the extra switch? I fail to see the difference to the original code here

Reply:
The difference is the failing else-part in the previous version.

Reply:
As discussed offline, we should split the individual sub-selects into different functions for better readability.